Section 3: While Loops
Problem 3.1: Reverse a Number (Medium)
Write a program that reverses the digits of a given number using a while loop.

Example: Input: 12345, Output: 54321

Hint: Initialize reverse=0. While n>0: get last digit (n%10), add to reverse (reverse=reverse*10+digit), remove last digit (n=n//10).

In [5]:
num=int(input("Enter your num:-"))
temp=num
rev=0
while temp>0:
    result=temp%10
    rev=(rev*10)+result
    temp=temp//10
print(f"orginal num {num} reverse num {rev}")




orginal num 121 reverse num 121


Problem 3.2: Sum of Digits (Easy)
Write a program that calculates the sum of all digits in a given number.

Example: Input: 12345, Output: 15 (1+2+3+4+5)

Hint: While n>0: add last digit to sum (sum += n%10), then remove it (n = n//10).


In [9]:
num=int(input("Enter you number:-"))
temp=num
rev=0
while temp>0:
    result=temp%10
    rev=rev+result
    temp=temp//10

print(f"number: {num}  sum of num :- {rev}")



number: 12345  sum of num :- 15



Problem 3.3: Number Guessing Game (Medium)
Write a number guessing game where:

The program generates a random number between 1 and 100
The user has to guess the number
After each guess, the program tells the user if their guess is too high or too low
The game continues until the user guesses correctly
Display the number of attempts taken
Hint: Use import random and random.randint(1, 100) to generate number. Use while loop with condition guess != target. Count attempts

In [1]:
import random

value= random.randint(1,100)

number=int(input("Guess a number between 1,100 :-"))

count=1

while value != number:

    if number>value:
        print(f"{number}:Guess Lower")
    else:
        print(f"{number}:Guess HIgher")

    number=int(input("Guess a number between 1,100 :-"))
    count+=1

print("You Guess Correct!!!!")
print(f"You Took {count} Attempt")

5:Guess HIgher
10:Guess Lower
7:Guess Lower
You Guess Correct!!!!
You Took 4 Attempt


Problem 3.4: Palindrome Checker (Medium)
Write a program to check if a given number is a palindrome or not.

A palindrome number reads the same forwards and backwards.

Examples: 121, 12321, 1331 are palindromes

Hint: Store original number, reverse it (like Problem 3.1), then compare original with reversed number.

In [13]:
number=int(input("Enter your number"))
temp=number
reverse=0

while temp>0:
    result=temp%10
    reverse=(reverse*10)+result
    temp=temp//10

if number==reverse:
    print("Palindrome")
else:
    print("Not a Palindrome")



Palindrome


Problem 3.5: Armstrong Number (Hard)
Write a program to check if a number is an Armstrong number (also called Narcissistic number).

An Armstrong number is a number that is equal to the sum of its own digits each raised to the power of the number of digits.

Examples:

153 = 1³ + 5³ + 3³ = 1 + 125 + 27 = 153 (Armstrong number)
9474 = 9⁴ + 4⁴ + 7⁴ + 4⁴ = 6561 + 256 + 2401 + 256 = 9474 (Armstrong number)
Hint: First count digits (power). Store original. Extract each digit, raise to power, add to sum. Compare sum with original.

In [26]:
num=(input("Enter your number"))
length=len(num)
value=int(num)
temp=value
answer=0
result=0
while temp>0:
    n=temp%10
    result=(n**length)
    answer=answer+result
    temp=temp//10

if value == answer:
    print(f"{answer} : Armstrong number")
else:
    print(f"{answer} : Not Armstrong number")




153 : Armstrong number
